src/packet_types.c

```diff
@@ -193,6 +193,36 @@ void acceleration_data_block_init(AccelerationDataBlock *b, const uint32_t measu
            &z_axis, sizeof(z_axis));
 }
 
+/**
+ * Initializes a telemetry data block with the provided information.
+ * @param b The telemetry request block to be initialized.
+ * @param data_subtype_1 The first requested data type.
+ * @param used_1 A boolean to determine if the first data request is valid.
+ * @param data_subtype_2 The second requested data type.
+ * @param used_2 A boolean to determine if the second data request is valid.
+ * @param data_subtype_3 The third requested data type.
+ * @param used_3 A boolean to determine if the third data request is valid.
+ * @param data_subtype_4 The fourth requested data type.
+ * @param used_4 A boolean to determine if the fourth data request is valid.
+ * */
+void telemetry_request_block_init(TelemetryRequestBlock *b, const uint8_t data_subtype_1,
+                             const uint8_t used_1, const uint8_t data_subtype_2,
+                             const uint8_t used_2, const uint8_t data_subtype_3,
+                             const uint8_t used_3, const uint8_t data_subtype_4,
+                             const uint8_t used_4) { 
+    memcpy(b->bytes, &data_subtype_1, sizeof(data_subtype_1));
+    memcpy(b->bytes + 1 + sizeof(data_subtype_1), &used_1, sizeof(data_subtype_1)); // + 1 and all further incrementations account for the dead space after the data subtype.
+    memcpy(b->bytes + 1 + sizeof(data_subtype_1) + sizeof(used_1), &data_subtype_2, sizeof(data_subtype_2));
+    memcpy(b->bytes + 2 + sizeof(data_subtype_1) + sizeof(used_1) + sizeof(data_subtype_2), &used_2, sizeof(used_2));
```

Comment:
In this case you can't use `memcpy` for everything. If you take a look at the spec, each byte contains the first 6 bits which is data subtype, a 7th unused bit, and an 8th bit which marks whether the request is valid.

You'll need to use bit shifting. Something like
```c
bytes[0] = (data_subtype_1 << 2);
bytes[0] |= (used_1);
// Repeat for bytes 1, 2 and 3
```

Reply:
When you use the |= operator, I understand that it does a bitwise OR operation, but I have no idea what that means. Documentation online is as readable as egyptian hieroglyphics translated back and forth between random languages on google translate. Is there any way you could ELI5? I understand there are some wacky concepts at play behind that so if you'd rather wait until the Thursday meeting I can do that too.

Reply:
Thank you! I'm still a little confused but I'll definitely take another crack at it before Thursday.